src/wfs/src/main/java/org/geoserver/wfs/kvp/FilterKvpParser.java

```diff
@@ -81,7 +82,9 @@ public Object parse(String value) throws Exception {
                 InputStream input = new ByteArrayInputStream(string.getBytes());
 
                 try {
-                    Filter filter = (Filter) parser.parse(input);
+                    InputSource inputSource = new InputSource(input);
+                    inputSource.setEncoding(Charset.defaultCharset().name());
```

Comment:
Hi Alvaro, I had a similar concern that I did not voice yet... which is that regardless of what the default OS charset is, the requests are coming from clients that are not running the local OS. I guess that at least for local administrations or services meant to be consumed by a single country, it could make sense to try and predict the charset, but yes, in general, it should be the client itself to add a charset indication in the filter to be parsed.